FlashFeetAPI/src/test/java/org/example/api/integration/AuthIntegrationTest.java

```diff
@@ -0,0 +1,120 @@
+package org.example.api.integration;
+
+import org.example.api.AuthService;
+import org.example.cli.Login;
+import org.example.db.AuthDao;
+import org.example.db.DatabaseConnector;
+import org.example.exception.FailedToLoginException;
+import org.example.exception.FailedTogenerateTokenException;
+import org.example.trueApplication;
+import org.junit.jupiter.api.DisplayName;
+import org.example.trueConfiguration;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.example.cli.Login;
+import io.dropwizard.configuration.ResourceConfigurationSourceProvider;
+import io.dropwizard.testing.junit5.DropwizardAppExtension;
+import io.dropwizard.testing.junit5.DropwizardExtensionsSupport;
+import org.junit.jupiter.api.Assertions;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * This class contains integration tests for the authentication feature.
+ */
+@ExtendWith(DropwizardExtensionsSupport.class)
+public class AuthIntegrationTest {
+
+    /**
+     * The application instance used for testing.
+     */
+    static final DropwizardAppExtension<trueConfiguration> APP = new DropwizardAppExtension<>(
+            trueApplication.class, null,
+            new ResourceConfigurationSourceProvider()
+    );
+
+    /**
+     * Tests if a valid login returns an OK response.
+     */
+    @Test
+    @DisplayName("Valid Login should return a OK response")
+    public void authLogin_ShouldAllowLogin_WhenValidLoginDetailsEntered() {
```

Comment:
these tests are more like unit tests. you should have 2 integration tests. 1 with valid login (use environment variables, don't harcode valid credentials even in tests) that expects a token to be returned, and 1 with invalid credentials that expects a 500 to be returned